matchms/Scores.py

```diff
@@ -49,7 +52,8 @@ class Scores:
         Cosine score between spectrum2 and spectrum3 is 0.14 with 1 matched peaks
         Cosine score between spectrum2 and spectrum4 is 0.61 with 1 matched peaks
     """
-    def __init__(self, references: ReferencesType, queries: QueriesType, similarity_function: SimilarityFunction):
+    def __init__(self, references: ReferencesType, queries: QueriesType,
+                 similarity_function: SimilarityFunction, is_symmetric: bool = False):
```

Comment:
Signature of SimilarityFunction is `Callable[[Sample, Sample], Score]` which is not what Scores wants, it requires object with pair and matrix methods. Please update type to BaseSimilarityFunction
```suggestion
                 similarity_function: BaseSimilarityFunction, is_symmetric: bool = False):
```